src/include/daos_uns.h

```diff
@@ -195,12 +195,15 @@ duns_destroy_path(daos_handle_t poh, const char *path);
 int
 duns_parse_attr(char *str, daos_size_t len, struct duns_attr_t *attr);
 
+int
+duns_create_attr(char *type, uuid_t pool, uuid_t cont, char **_value, daos_size_t *_out_size);
```

Comment:
@mchaarawi @bfaccini  I don't really want to make this function public but there's no "duns_internal.h", my other option is to put it in "dfs_internal.h" if you think this would be OK.

Reply:
yes this should not be public and mostly because you use D_ASPRINTF, and then you will expect the user to call D_FREE, which they can't.
but why you need this to be in the UNS? can't you add this function just internally in dfuse?
all you need is to move DUNS_XATTR_FMT def to this header. i think that would be ok?

otherwise maybe just allocate the value buffer outside of this function, and maybe call it something:
duns_format_xattr ?

Reply:
Do we want DUNS_XATTR_FMT to be public?  That's why the code needed to be in duns.c to get access to that define.  Perhaps we could put DUNS_XATTR_FMT in dfs_internal.h?

Reply:
I don't see a major reason not to make it public..

i would really like to split the duns outside of dfs. so i don't think we should put it in dfs_internal.h.

Reply:
Done.  Like you say there's some layout changes that would be beneficial here.